src/CondenserDotNet.Client/Configuration/JsonKeyValueParser.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using CondenserDotNet.Client.DataContracts;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace CondenserDotNet.Client.Configuration
+{
+    public class JsonKeyValueParser : IKeyParser
+    {
+        private readonly Stack<string> _context = new Stack<string>();
+
+        private readonly IDictionary<string, string> _data =
```

Comment:
Seeing as it isn't exposed should we not just use the concrete type rather than the interface.
http://www.nimaara.com/2016/03/06/beware-of-the-idictionary-tkey-tvalue/
is an interesting read around the IDictionary interface returning the non generic enumerator. Small perf improvement